### ISSUE TITLE
[ISSUE] Existing stopped Databricks cluster with Libraries installed fails refresh with library status exception

### ISSUE THREAD

**Issue by jamescross91** [databrickslabs/terraform-provider-databricks#952]
Hi there,

Thank you for opening an issue. Please note that we try to keep the Databricks Provider issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

### Configuration
```hcl
esource "databricks_cluster" "interactive_cluster" {
  cluster_name            = var.databricks_cluster_name
  spark_version           = var.databricks_cluster_spark_version
  node_type_id            = var.databricks_cluster_node_type_id
  idempotency_token       = random_string.idempotency_token.result
  autotermination_minutes = 480
  is_pinned               = true   
  autoscale {
    min_workers = 1
    max_workers = 20
  }
  azure_attributes {
    availability       = "ON_DEMAND_AZURE"
    first_on_demand    = 1
    spot_bid_max_price = -1
  }

  library {
    pypi {
      package = "kedro"
    }
  }
  library {
    pypi {
      package = "plotly"
    }
  }
  library {
    pypi {
      package = "azure-identity==1.5.0"
    }
  }
  library {
    pypi {
      package = "azure-storage-blob==12.6.0"
    }
  }
  library {
    pypi {
      package = "pyodbc==4.0.30"
    }
  }
  library {
    pypi {
      package = "boxsdk==2.11.0"
    }
  }
  library {
    pypi {
      package = "azure-keyvault-secrets==4.2.0"
    }
  }
}
```

### Expected Behavior
Refresh shows library status

### Actual Behavior
Error: cannot read cluster: 0 libraries are ready, but there are still 7 pending

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform plan`

### Terraform and provider versions

Terraform version: 1.0.3, Databricks 0.4.0

### Debug Output

```
2021-12-10T12:27:45.059Z [DEBUG] provider.terraform-provider-databricks_v0.4.0: GET /api/2.0/libraries/cluster-status: timestamp=2021-12-10T12:27:45.059Z
2021-12-10T12:27:45.251Z [DEBUG] provider.terraform-provider-databricks_v0.4.0: 200 OK {
  "cluster_id": "0310-154206-pined743",
  "library_statuses": [
    {
      "is_library_for_all_clusters": false,
      "library": {
        "pypi": {
          "package": "azure-keyvault-secrets==4.2.0"
        }
      },
      "status": "PENDING"
    },
    {
      "is_library_for_all_clusters": false,
      "library": {
        "pypi": {
          "package": "azure-storage-blob==12.6.0"
        }
      },
      "status": "PENDING"
    },
    {
      "is_library_for_all_clusters": false,
      "library": {
        "pypi": {
          "package": "boxsdk==2.11.0"
        }
      },
      "status": "PENDING"
    },
    {
      "is_library_for_all_clusters": false,
      "library": {
        "pypi": {
          "package": "azure-identity==1.5.0"
        }
      },
      "status": "PENDING"
    },
    {
      "is_library_for_all_clusters": false,
      "library": {
        "pypi": {
          "package": "plotly"
        }
      },
      "status": "PENDING"
    },
```

### Important Factoids
Works fine with databricks provider version 0.3.11, or if you manually start the cluster up before/during a refresh, otherwise it endlessly polls the libraries/cluster-status API, and because the cluster isn't started the status reports as pending.  Refresh should accept pending state as valid.

**Comment by moredatapls** [comment]
I have the same issue after I tried to upgrade to 0.4.0, also worked for me in 0.3.*

**Comment by nfx** [comment]
This fix will be part of 0.4.1, which will be released shortly
